bundles/framework/featuredata2/plugin/MapSelectionPlugin.ol3.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * @class Oskari.mapframework.bundle.metadata.plugin.MapSelectionPlugin
+ *
+ * Provides functionality to draw a selection box on the map
+ */
+Oskari.clazz.define(
+    'Oskari.mapframework.bundle.featuredata2.plugin.MapSelectionPlugin',
+    function (config, sandbox) {
+        var me = this;
+
+        me._clazz =
+            'Oskari.mapframework.bundle.featuredata2.plugin.MapSelectionPlugin';
+        me._name = 'MapSelectionPlugin';
+        if(config.instance){
+            me.caller = config.instance;
+        }
+        me.drawControls = null;
+        me.editMode = false;
+        me.listeners = [];
+        me.currentDrawMode = null;
+        me.prefix = 'Default.';
+        me.sandbox = sandbox;
+        me.WFSLayerService = me.sandbox.getService('Oskari.mapframework.bundle.mapwfs2.service.WFSLayerService');
+        me._currentRequestID = null;
+        me._features;
+        me._drawing;
+
+        if (me._config && me._config.id) {
+            me.prefix = me._config.id + '.';
+        }
+        // graphicFill, instance
+        if (me._config && me._config.graphicFill) {
+            me.graphicFill = me._config.graphicFill;
+        }
+        me.multipart = (me._config && me._config.multipart === true);
+    }, {
+
+        /**
+         * @method addListener
+         * Registers a listener that will be notified when a selection has been made.
+         * The function will receive the selection geometry as parameter (OpenLayers.Geometry).
+         * @param {Function} listenerFunction
+         */
+        addListener: function (listenerFunction) {
```

Comment:
Should be replaced with Oskari.makeObservable(this) and callers using plugin.on('eventname', listenerFunction)